src/CoreFoundation/CFProxySupport.cs

```diff
@@ -579,62 +581,328 @@ public static CFProxySettings GetSystemProxySettings ()
 			dict.DangerousRelease ();
 			return new CFProxySettings (dict);
 		}
-		
-#if notyet
-		// FIXME: These require bindings for CFRunLoopSource and CFStreamClientContext
-		
-		public delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, NSArray proxyList, NSError error);
-		
+
+		delegate void CFProxyAutoConfigurationResultCallbackInternal (IntPtr client, IntPtr proxyList, IntPtr error);
+		public delegate void CFProxyAutoConfigurationResultCallback (NSObject client, CFProxy[] proxyList, NSError error);
+		// helper delegate to reuse code
+		delegate IntPtr CreatePACCFRunLoopSource (CFProxyAutoConfigurationResultCallbackInternal cb, ref CFStreamClientContext context);
+
+		static CFProxy[] ParseProxies (IntPtr proxyList)
+		{
+			CFProxy[] proxies = null;
+			if (proxyList != IntPtr.Zero) {
+				// it was retained in the cbs.
+				using (var array = new CFArray (proxyList, true)) {
+					proxies = new CFProxy [array.Count];
+					for (int i = 0; i < proxies.Length; i++) {
+						var dict = new NSDictionary (array.GetValue (i));
```

Comment:
I think you should use `Runtime.GetNSObject (array.GetValue (i))` instead of manually creating the `NSDictionary` using the pointer. That works better if the same native pointer shows up multiple times.